server/enterprise/message_export/global_relay_export/global_relay_export_test.go

```diff
@@ -1285,25 +1287,26 @@ func TestGlobalRelayExport(t *testing.T) {
 				}
 			}
 
-			dest, err := os.CreateTemp("", "")
+			exportFileName := path.Join("export", "jobName", "jobName-batch001.zip")
+			results, err := GlobalRelayExport(rctx, Params{
+				ExportType:            model.ComplianceExportTypeGlobalrelayZip,
+				Posts:                 tt.posts,
+				BatchPath:             exportFileName,
+				Db:                    shared.NewMessageExportStore(mockStore),
+				FileAttachmentBackend: fileBackend,
+				ExportBackend:         exportBackend,
+				Templates:             templatesContainer,
+			})
 			assert.NoError(t, err)
-			defer os.Remove(dest.Name())
+			assert.Equal(t, tt.expectedWarnings, results.NumWarnings)
 
-			_, warningCount, err := GlobalRelayExport(rctx, tt.posts, shared.NewMessageExportStore(mockStore), fileBackend, dest, templatesContainer)
+			zipBytes, err := exportBackend.ReadFile(exportFileName)
 			assert.NoError(t, err)
-			assert.Equal(t, tt.expectedWarnings, warningCount)
-
-			_, err = dest.Seek(0, 0)
-			assert.NoError(t, err)
-
-			destInfo, err := dest.Stat()
-			assert.NoError(t, err)
-
-			zipFile, err := zip.NewReader(dest, destInfo.Size())
+			zipReader, err := zip.NewReader(bytes.NewReader(zipBytes), int64(len(zipBytes)))
 			assert.NoError(t, err)
 
 			if tt.expectedFiles > 0 {
-				firstFile, err := zipFile.File[0].Open()
+				firstFile, err := zipReader.File[0].Open()
```

Comment:
Does the "file" need closing?

Reply:
I'm not totally sure, tbh. It'll get picked up by the gc, and it's in a test... But may as well be complete. I'll add that to the most recent PR I'm working on because it's redoing a lot of these tests and I want to avoid merge hell. 
Thanks!